pkg/apis/core/types.go

```diff
@@ -5788,3 +5796,470 @@ type PortStatus struct {
+	// when creating it. No other fields are allowed and will be rejected during
+	// validation.
+	metav1.ObjectMeta
+
+	// The specification for the ResourceClaim. The entire content is
+	// copied unchanged into the PVC that gets created from this
+	// template. The same fields as in a ResourceClaim are also valid here.
+	Spec ResourceClaimSpec
+}
+
+// ResourceClaimSpec defines how a resource is to be allocated.
+type ResourceClaimSpec struct {
+	// ResourceClassName references the driver and additional parameters
+	// via the name of a ResourceClass that was created as part of the
+	// driver deployment.
+	//
+	// The apiserver does not check that the referenced class exists, but a
+	// driver-specific admission webhook may require that and is allowed to
+	// reject claims where the class is missing.
+	ResourceClassName string
+
+	// ParametersRef references a separate object with arbitrary parameters
+	// that will be used by the driver when allocating a resource for the
+	// claim.
+	//
+	// The object must be in the same namespace as the ResourceClaim.
+	ParametersRef *ResourceClaimParametersReference
+
+	// Allocation can start immediately or when a Pod wants to use the
+	// resource. Waiting for a Pod is the default.
+	AllocationMode AllocationMode
+}
+
+// AllocationMode describes whether a ResourceClaim gets allocated immediately
+// when it gets created (AllocationModeImmediate) or whether allocation is
+// delayed until it is needed for a Pod
+// (AllocationModeWaitForFirstConsumer). Other modes might get added in the
+// future.
+type AllocationMode string
+
+const (
+	// When a ResourceClaim has AllocationModeWaitForFirstConsumer, allocation is
+	// delayed until a Pod gets scheduled that needs the ResourceClaim. The
+	// scheduler will consider all resource requirements of that Pod and
+	// trigger allocation for a node that fits the Pod.
+	AllocationModeWaitForFirstConsumer AllocationMode = "WaitForFirstConsumer"
+
+	// When a ResourceClaim has AllocationModeImmediate, allocation starts
+	// as soon as the ResourceClaim gets created. This is done without
+	// considering the needs of Pods that will use the ResourceClaim
+	// because those Pods are not known yet.
+	AllocationModeImmediate AllocationMode = "Immediate"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ResourceClaim is created by users to describe which resources they need.
+// Its status tracks whether the resource has been allocated and what the
+// resulting attributes are.
+type ResourceClaim struct {
+	metav1.TypeMeta
+
+	// The driver must set a finalizer here before it attempts to allocate
+	// the resource. It removes the finalizer again when a) the allocation
+	// attempt has definitely failed or b) when the allocated resource was
+	// freed. This ensures that resources are not leaked.
+	//
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
```

Comment:
These are k8s developer docs, not k8s user docs.

Reply:
So you want me to remove the link? Done.

Reply:
Not sure what you wanted to document, but you could maybe link to the API reference docs? Although I think the generated documentation will have a link to the documentation for ObjectMeta already.

Reply:
I think I just wanted to provide more information about ObjectMeta because the type itself doesn't have good documentation. But other types don't do that either, so it should be fine to just do the same also for the new types.